internals/__init__.py

```diff
@@ -53,6 +53,13 @@ def __init__(self, delta_energy, delta_fulfillment):
 }
 
 
+def advance_time():
```

Comment:
maybe rename this, since its a generator not a function which advances the time

Reply:
Addressed in dc42261